apps/api/src/app/inbox/usecases/bulk-update-preferences/bulk-update-preferences.usecase.ts

```diff
@@ -0,0 +1,111 @@
+import { Injectable, NotFoundException, BadRequestException, UnprocessableEntityException } from '@nestjs/common';
+import { AnalyticsService, InstrumentUsecase } from '@novu/application-generic';
+import {
+  NotificationTemplateEntity,
+  NotificationTemplateRepository,
+  SubscriberRepository,
+  BaseRepository,
+} from '@novu/dal';
+import { PreferenceLevelEnum } from '@novu/shared';
+
+import { AnalyticsEventsEnum } from '../../utils';
+import { InboxPreference } from '../../utils/types';
+import { BulkUpdatePreferencesCommand } from './bulk-update-preferences.command';
+import { UpdatePreferences } from '../update-preferences/update-preferences.usecase';
+import { UpdatePreferencesCommand } from '../update-preferences/update-preferences.command';
+import { BulkUpdatePreferenceItemDto } from '../../dtos/bulk-update-preferences-request.dto';
+
+const MAX_BULK_LIMIT = 100;
+
+@Injectable()
+export class BulkUpdatePreferences {
+  constructor(
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private subscriberRepository: SubscriberRepository,
+    private analyticsService: AnalyticsService,
+    private updatePreferencesUsecase: UpdatePreferences
+  ) {}
+
+  @InstrumentUsecase()
+  async execute(command: BulkUpdatePreferencesCommand): Promise<InboxPreference[]> {
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+    if (!subscriber) throw new NotFoundException(`Subscriber with id: ${command.subscriberId} is not found`);
+
+    if (command.preferences.length === 0) {
+      throw new BadRequestException('No preferences provided for bulk update');
+    }
+
+    if (command.preferences.length > MAX_BULK_LIMIT) {
+      throw new UnprocessableEntityException(`Exceeded maximum limit of ${MAX_BULK_LIMIT} preferences for bulk update`);
+    }
+
+    const allWorkflowIds = command.preferences.map((preference) => preference.workflowIdOrInternalId);
+    const workflowInternalIds = allWorkflowIds.filter((id) => BaseRepository.isInternalId(id));
+    const workflowIdentifiers = allWorkflowIds.filter((id) => !BaseRepository.isInternalId(id));
+
+    const dbWorkflows = await this.notificationTemplateRepository.find({
+      _environmentId: command.environmentId,
+      $or: [{ _id: { $in: workflowInternalIds } }, { 'triggers.identifier': { $in: workflowIdentifiers } }],
+    });
```

Comment:
split the mongo ids from the identifiers, this ensures that we don't get the mongo casting error to ObjectId